### PR TITLE
docs(examples) Restructure documentation to add LIN

### PR DESCRIPTION
**Pull request by bcornish**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nixnet-python/blob/master/CONTRIBUTING.rst).
- [x] New tests have been created for any new features or regression tests for bugfixes.
- [x] `tox` successfully runs, including unit tests and style checks (see [CONTRIBUTING.md](https://github.com/ni/nixnet-python/blob/master/CONTRIBUTING.rst)).

### What does this Pull Request accomplish?

Add diff and new headings for example documentation. 

### Why should this Pull Request be merged?

Add LIN documentation.

### CODE REVIEW

**Comment by bcornish** [comment]
It looked strange to me to have the diff header under several sections. Instead, I created a diff example doc on its own and then referenced that within the different types of IO examples.

File: docs/examples/queued_io.rst [hunk 1]
```diff
@@ -5,13 +5,21 @@ This example uses :any:`nixnet.session.FrameInQueuedSession` and
...
       Chooses the frame object to create based on the ``frame_type`` field in
       the raw data.  This can be overridden by passing a custom
       :any:`nixnet.types.FrameFactory` in the ``frame_type`` parameter.
+   ``change_lin_sched``:
+      Writes a request for the LIN interface to change
+      the running schedule.
 
-.. literalinclude:: ../../nixnet_examples/can_frame_queued_io.py
+The CAN and LIN Stream I/O examples are similar. Use this diff to compare: :doc:`can_lin_diff` 
    :pyobject: main
```
Comment (epage) [inline docs/examples/queued_io.rst]: Does this line still make sense?

File: docs/examples/queued_io.rst [hunk 2]
```diff
@@ -5,13 +5,21 @@ This example uses :any:`nixnet.session.FrameInQueuedSession` and
 :any:`nixnet.session.FrameOutQueuedSession` to demonstrate how queued sessions
 work.
 
+CAN Queued I/O
+---------------
+
+.. literalinclude:: ../../nixnet_examples/can_frame_queued_io.py
+
 To adapt this example to LIN frames, reference signals in a database that use LIN:
```
Comment (epage) [inline docs/examples/queued_io.rst]: Personally, I'd recommend making the new rst file be about adapting CAN examples for LIN and include all of this text.

File: docs/examples/single_point_io.rst [hunk 3]
```diff
@@ -10,7 +10,13 @@ To adapt this to Frames, just change the sessions to
 :any:`nixnet.session.FrameOutSinglePointSession` with frames instead of
 signals.  Then adjust ``read``/``write`` to take a frame object per frame
 configured in the session rather than signals.  This works for both CAN and LIN
-frames.  See :doc:`queued_io` to see how to read and write frames.
+frames. LIN frames also require ``change_lin_sched`` to write a request for the
+LIN interface to change the running schedule  See :doc:`queued_io` to see how
+to read and write frames. See :doc:`can_lin_diff` to see how to convert a CAN
+example into a LIN example.
```
Comment (epage) [inline docs/examples/single_point_io.rst]: Is it sufficient to point them to `queued_io` and rely on that example pointing to the document "how to adapt CAN to LIN"?

File: docs/examples/stream_io.rst [hunk 4]
```diff
@@ -1,17 +1,30 @@
...
+      the running schedule.
 
-.. literalinclude:: ../../nixnet_examples/can_frame_stream_io.py
+LIN Stream I/O
+--------------
+
+.. literalinclude:: ../../nixnet_examples/lin_frame_stream_io.py
+
+The CAN and LIN Stream I/O examples are similar. Use this diff to compare: :doc:`can_lin_diff` 
    :pyobject: main
```
Comment (epage) [inline docs/examples/stream_io.rst]: Again, not sure if `:pyobjject:` applies here

**Comment by bcornish** [comment]
I add changes to reference the CAN/ LIN diff that I renamed 'Adapting CAN examples to LIN'

**Comment by bcornish** [comment]
Removed LIN specific changes from IO examples.

File: docs/examples/queued_io.rst [hunk 5]
```diff
@@ -5,13 +5,10 @@ This example uses :any:`nixnet.session.FrameInQueuedSession` and
...
-      Chooses the frame object to create based on the ``frame_type`` field in
-      the raw data.  This can be overridden by passing a custom
-      :any:`nixnet.types.FrameFactory` in the ``frame_type`` parameter.
+CAN Queued I/O
+---------------
 
 .. literalinclude:: ../../nixnet_examples/can_frame_queued_io.py
-   :pyobject: main
+
+Refer to :doc:`can_lin_diff` for a diff of CAN and LIN examples.
```
Comment (epage) [inline docs/examples/queued_io.rst]: From
> for a diff of CAN and LIN examples.

to
> for how to adapt from CAN to LIN.

File: docs/examples/stream_io.rst [hunk 6]
```diff
@@ -1,17 +1,18 @@
...
 
 .. literalinclude:: ../../nixnet_examples/can_frame_stream_io.py
-   :pyobject: main
+
+LIN Stream I/O
+--------------
+
+.. literalinclude:: ../../nixnet_examples/lin_frame_stream_io.py
+
+Refer to :doc:`can_lin_diff` for a diff of CAN and LIN examples.
```
Comment (epage) [inline docs/examples/stream_io.rst]: From
> for a diff of CAN and LIN examples.

to
> for how to adapt from CAN to LIN.

File: docs/examples/can_lin_diff.rst [hunk 7]
```diff
@@ -0,0 +1,20 @@
+Adapting CAN examples to LIN
+============================
+
+This displays the diff of ``can_frame_stream_io.py`` and 
+``lin_frame_stream_io.py`` to demonstrate the changes required to
+update CAN example code for LIN.
+
+To adapt this example to LIN frames, reference signals in a database that use LIN:
```
Comment (epage) [inline docs/examples/can_lin_diff.rst]: 1. "to adapt the examples from CAN to LIN"...
2. This should go before the diff

Comment (bcornish) [inline docs/examples/can_lin_diff.rst]: It is before the diff? Unless I'm misunderstanding.

File: docs/examples/queued_io.rst (hunk 5 above)
Comment (bcornish) [inline docs/examples/queued_io.rst]: Done.

File: docs/examples/can_lin_diff.rst (hunk 7 above)
Comment (bcornish) [inline docs/examples/can_lin_diff.rst]: Yep, I was not understanding. Thanks for clearing that up @epage. Added these in my latest commit. Sorry for the oversights!

**Pull request merged**